src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -222,13 +222,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
 
         Private Function ScanNextCharAsToken(leadingTrivia As SyntaxList(Of VisualBasicSyntaxNode)) As SyntaxToken
             Dim token As SyntaxToken
-
-            If Not CanGet() Then
+            Dim c As Char
+            If Not Peep(c) Then
```

Comment:
The easiest is to try debugging Release with JIT optimizations not suppressed (setting under ToolsOptionsDebug) and take a look at x86/x64 at the calling site. 
Looking at the actual implementation of Peek it seems a bit too large to be an attractive inlining target.
As I commented above, For the cases where CanGet is a condition for the following Peek, folding that into a single Peep seems ok (since we assume that CanGet is mostly true). For the cases where the char is not used (like in negative tests), CanGet is likely to be substantially cheaper.